src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,329 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Module to perform Trotter-Suzuki decompositions to output as circuits."""
+
+from openfermion.ops import QubitOperator
+from openfermion.utils import count_qubits
+import numpy
+
+"""
+Description:
+    Functions for estimating the exponential of an operator
+    composed of Pauli matrices, by Trotterization. Outputs QASM
+    format to a python stream.
+
+    Change-of-basis gates:
+    H           : Z to X basis (and back)
+    Rx(pi/2)    : Z to Y basis .....
+    Rx(-pi/2)   : Y to Z basis .....
+"""
+
+
+def third_order_trotter_helper(op_a, op_b):
+    """Recursively trotterize a QubitOperator according
+        to the scheme: e^(A+B) = e^(7/24 * A) e^(2/3 * B) e^(3/4 * A)
+        e^(-2/3 * B) e^(-1/24 * A) e^(B)
+
+    Note:
+        See N. Hatano and M. Suzuki Lect. Notes Phys 679, 37-68 (2005)
+
+    Args:
+        op_a: (QubitOperator) first term in hamiltonian
+        op_b: (list of QubitOperator) the rest of the terms in the
+                hamiltonian as a list of QubitOperators
+    Returns:
+        list of QubitOperators giving the trotterized hamiltonian
+    """
+    # Create list to hold return value
+    ret_val = []
+
+    # e^(7/24 * A)
+    ret_val.append(7.0/24.0 * op_a)
+
+    # e^(2/3 * B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(2.0 / 3.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(op * 2.0/3.0)
+        atemp = op_b[0] * 2.0/3.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    # e^(3/4 * A)
+    ret_val.append(3.0/4.0 * op_a)
+
+    # e^(-2/3 * B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(-2.0/3.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(op * -2.0/3.0)
+        atemp = op_b[0] * -2.0/3.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    # e^(-1/24 * A)
+    ret_val.append(-1.0/24.0 * op_a)
+
+    # e^(B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(1.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(1.0 * op)
+        atemp = op_b[0] * 1.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    return ret_val
+
+
+def get_trotterized_qubops(
+        hamiltonian,
+        trotter_number=None,
+        trotter_order=None,
+        term_ordering=None,
+        k_exp=None):
+    """Perform Trotter decomposition without exponentiating, to be
```

Comment:
Style wise we try to keep the short description in a single line, with optional longer description below.  However, from this description I can't tell what this function does.  Trotter is only defined for exponential operator splitting, so what is being computed?

Reply:
After reading through the routine, a better name might be something like trotter_operator_grouping, or trotter_operator_helper, because no actual trotter is performed, it's just a way of grouping operators that is associated with Trotter.  This will make it difficult to distinguish from the routine that actually does the exponentiation.

Reply:
If possible, write in the description a fairly precise description of what happens in this function.  For example, I don't think -1.0j is added here, so the output Hamiltonian is still hermitian.